rocAL/rocAL/source/label_reader_folders.cpp

```diff
@@ -142,14 +142,15 @@ void LabelReaderFolders::read_all(const std::string& _path)
         }
         else if(filesys::exists(pathObj) && filesys::is_directory(pathObj))
         {
+            dir_counter++;
             _folder_path = subfolder_path;
             _subfolder_file_names.clear();
             read_files(_folder_path);
             for(unsigned i = 0; i < _subfolder_file_names.size(); i++) {
-                add(_subfolder_file_names[i], dir_count);
+                add(_subfolder_file_names[i], dir_counter);
```

Comment:
I am not sure if this is the right fix for the issue. I think you want to increase the dir_count only for valid directories. The new code will put label =0 for all the subfolders which is not correct